text/0774-implicit-record-route-loading.md

```diff
@@ -0,0 +1,89 @@
+---
+stage: accepted
+start-date: 2021-11-14
+release-date: 
+release-versions:
+teams: # delete teams that aren't relevant
+  - framework
+prs:
+  accepted: https://github.com/emberjs/rfcs/pull/774 
+---
+
+# Deprecate Implicit Record Loading in Routes
+
+## Summary
+
+This RFC seeks to deprecate and remove the default record loading behaviour on Ember's Route. By consequence, this will also deprecate and remove the default store on every Route.  This behaviour is likely something you either know you have or do not know but it may be occuring in your app.
+
+```js
+export default class PostRoute extends Route {
+  beforeModel() {
+    // do stuff
+  }
+
+  afterModel() {
+    // do stuff
+  }
+}
+```
+
+In this example, the `model` hook is not defined.  However, Ember will attempt to try a few things before rendering this route's template.
+
+1. If there is a `store` property on your route, it will attempt to call it's `find` method.  Assuming you have `ember-data` installed, you may be expecting this. The arguments will be extracted from the params.
+  a. For example, if a dynamic segment is `:post_id`, there exists logic to split on the underscore and find a record of type `post`.  
+
+2. As a fallback, it will attempt to define a `find` method and use your `Model` instance's find method to fetch.  If no `find` method exists, an assertion is thrown.
```

Comment:
```suggestion
2. As a fallback, it will attempt to define a `find` method and use your `Model` instance's find method to fetch.  If a `Model` cannot be found or if the found `Model` does not have a `find` method, an assertion is thrown.
```